ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/switch_data.py

```diff
@@ -47,8 +48,9 @@ def switch_data(self: SharedUtils) -> dict:
         node_config = {}
         hostname = self.hostname
 
+        # Loading data from node_type_key (ex. l3leaf)
         node_type_key = self.node_type_key_data["key"]
-        node_type_config = get(self.hostvars, f"{node_type_key}", required=True)
+        node_type_config = get(self.hostvars, f"{node_type_key}", default={})
```

Comment:
so if my understanding is correct we now can ignore the requirement for having a node_type_key defined in our hostvars at this stage?

Does it have an impact in other places of AVD ? is type still required?